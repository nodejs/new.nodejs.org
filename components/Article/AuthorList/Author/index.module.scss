.link {
  &:hover img {
    transform: scale(1.1);
  }

  img {
    background-size: contain;
    border: 2px solid var(--brand-light);
    border-radius: 100%;
    display: block;
    height: 30px;
    margin-top: 5px;
    transition: all 0.2s ease-in-out;
    width: 30px;
  }
}
