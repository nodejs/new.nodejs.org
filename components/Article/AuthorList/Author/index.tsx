import { useState } from 'react';
import { useIntl } from 'react-intl';
import Image from 'next/image';
import styles from './index.module.scss';
import type { FC } from 'react';

type AuthorProps = { username: string; size?: number };

const Author: FC<AuthorProps> = ({ username, size }) => {
  const githubUserName = username.trim();
  const githubLink = `https://github.com/${githubUserName}`;
  const githubImgLink = `https://github.com/${githubUserName}.png?size=${size}`;

  const intl = useIntl();

  const [authorImg, setAuthorImg] = useState(githubImgLink);

  const translation = intl.formatMessage(
    { id: 'components.article.author.githubLinkLabel' },
    { username }
  );

  return (
    <a
      className={styles.link}
      href={githubLink}
      aria-label={translation}
      key={username}
      target="_blank"
      rel="noopener noreferrer"
    >
      <Image
        alt={githubUserName}
        src={authorImg}
        placeholder="blur"
        blurDataURL="/placeholder-img.png"
        width={size}
        height={size}
        onError={() => setAuthorImg('/placeholder-img.png')}
      />
    </a>
  );
};

export default Author;
