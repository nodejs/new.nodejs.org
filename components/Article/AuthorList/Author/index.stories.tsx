import Author from './index';
import type { Meta as MetaObj, StoryObj } from '@storybook/react';

type Story = StoryObj<typeof Author>;
type Meta = MetaObj<typeof Author>;

export const Default: Story = {
  args: {
    username: 'nodejs',
    size: 60,
  },
};

export const WithourUsername: Story = {
  args: {
    username: '',
    size: 0,
  },
};

export default { component: Author } as Meta;
