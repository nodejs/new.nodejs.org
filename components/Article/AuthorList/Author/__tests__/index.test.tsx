import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import Author from '..';

describe('Author component', () => {
  it('renders correctly', () => {
    const username = 'test-author';
    const { container } = render(
      <IntlProvider locale="en" onError={() => {}}>
        <Author username={username} size={60} />
      </IntlProvider>
    );
    expect(container).toMatchSnapshot();
  });

  it('does not render without a username', () => {
    const { container } = render(
      <IntlProvider locale="en" onError={() => {}}>
        <Author username="" size={0} />
      </IntlProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
