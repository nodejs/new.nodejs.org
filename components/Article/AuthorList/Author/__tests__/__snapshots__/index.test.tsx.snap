// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author component does not render without a username 1`] = `
<div>
  <li>
    <a
      aria-label="components.article.author.githubLinkLabel"
      class="link"
      href="https://github.com/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt=""
        data-nimg="1"
        decoding="async"
        height="0"
        loading="lazy"
        src="http://localhost/_next/image?url=https%3A%2F%2Fgithub.com%2F.png%3Fsize%3D0&w=16&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fgithub.com%2F.png%3Fsize%3D0&w=16&q=75 1x"
        style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
        width="0"
      />
    </a>
  </li>
</div>
`;

exports[`Author component renders correctly 1`] = `
<div>
  <li>
    <a
      aria-label="components.article.author.githubLinkLabel"
      class="link"
      href="https://github.com/test-author"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="test-author"
        data-nimg="1"
        decoding="async"
        height="60"
        loading="lazy"
        src="http://localhost/_next/image?url=https%3A%2F%2Fgithub.com%2Ftest-author.png%3Fsize%3D60&w=128&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fgithub.com%2Ftest-author.png%3Fsize%3D60&w=64&q=75 1x, /_next/image?url=https%3A%2F%2Fgithub.com%2Ftest-author.png%3Fsize%3D60&w=128&q=75 2x"
        style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
        width="60"
      />
    </a>
  </li>
</div>
`;
